python/test/gluon/test_lowerings.py

```diff
@@ -811,3 +811,127 @@ def kernel(x_ptr, y_ptr, shape_tuple: ttgl.constexpr, src_layout: ttgl.constexpr
+                    lane_bases=[[0, 8], [8, 0], [1, 0], [4, 0], [16, 0]],
+                    warp_bases=[[0, 1], [0, 4]],
+                    block_bases=[],
+                    shape=[128, 64],
+                ),
+                ttgl.DistributedLinearLayout(
+                    reg_bases=[[0, 2], [32, 0], [0, 32], [2, 0], [0, 16], [64, 0], [128, 0]],
+                    lane_bases=[[0, 8], [8, 0], [1, 0], [4, 0], [16, 0]],
+                    warp_bases=[[0, 1], [0, 4]],
+                    block_bases=[],
+                    shape=[256, 64],
+                ),
+            ),
+        ]
+    elif THREADS_PER_WARP == 64:
+        return [
+            (
+                0,
+                ttgl.DistributedLinearLayout(
+                    reg_bases=[[0, 2], [2, 0]],
+                    lane_bases=[[0, 8], [8, 0], [1, 0], [4, 0], [16, 0], [32, 0]],
+                    warp_bases=[[0, 1], [0, 4]],
+                    block_bases=[],
+                    shape=[64, 16],
+                ),
+                ttgl.DistributedLinearLayout(
+                    reg_bases=[[2, 0], [0, 2]],
+                    lane_bases=[[0, 8], [16, 0], [1, 0], [8, 0], [4, 0], [32, 0]],
+                    warp_bases=[[0, 1], [0, 4]],
+                    block_bases=[],
+                    shape=[64, 16],
+                ),
+            ),
+            (
+                0,
+                ttgl.DistributedLinearLayout(
+                    reg_bases=[[0, 2], [2, 0], [0, 16], [0, 32]],
+                    lane_bases=[[0, 8], [8, 0], [1, 0], [4, 0], [16, 0], [32, 0]],
+                    warp_bases=[[0, 1], [0, 4]],
+                    block_bases=[],
+                    shape=[64, 64],
+                ),
+                ttgl.DistributedLinearLayout(
+                    reg_bases=[[0, 2], [0, 32], [2, 0], [0, 16], [64, 0]],
+                    lane_bases=[[0, 8], [8, 0], [1, 0], [4, 0], [16, 0], [32, 0]],
+                    warp_bases=[[0, 1], [0, 4]],
+                    block_bases=[],
+                    shape=[128, 64],
+                ),
+            ),
+        ]
+    else:
+        return []
+
+
+@pytest.mark.parametrize("axis, src_layout, index_layout", _gather_layouts())
+def test_gather_layouts(axis, src_layout, index_layout, device):
+    src_shape = src_layout.shape
+    indices_shape = index_layout.shape
+
+    @gluon.jit
+    def kernel(
+        src_ptr,
+        idx_ptr,
+        out_ptr,
+        axis: ttgl.constexpr,
+        src_dim0: ttgl.constexpr,
+        src_dim1: ttgl.constexpr,
+        idx_dim0: ttgl.constexpr,
+        idx_dim1: ttgl.constexpr,
+        src_layout: ttgl.constexpr,
+        idx_layout: ttgl.constexpr,
+    ):
+        offs_src_dim0 = ttgl.arange(0, src_dim0, layout=ttgl.SliceLayout(1, src_layout))[:, None]
+        offs_src_dim1 = ttgl.arange(0, src_dim1, layout=ttgl.SliceLayout(0, src_layout))[None, :]
+        src_offs = offs_src_dim0 * src_dim1 + offs_src_dim1
```

Comment:
Not for this PR, but I think that this pattern is common enough that we might want to create some sugar for it. cc @peterbell10